crates/common/tedge_config_macros/examples/macro.rs

```diff
@@ -15,6 +15,45 @@ pub enum ReadError {
     GenericError(String),
 }
 
+pub trait AppendRemoveItem {
+    type Item;
+
+    fn append(current_value: &mut Option<Self::Item>, new_value: Self::Item);
+
+    fn remove(current_value: &mut Option<Self::Item>, remove_value: Self::Item);
+}
+
+macro_rules! append_remove_for_single_value {
+    ($($type:ty),*) => {
+        $(
+            impl AppendRemoveItem for $type {
+                type Item = $type;
+                fn append(current_value: &mut Option<Self::Item>, new_value: Self::Item) {
+                    *current_value = Some(new_value);
+                }
+
+                fn remove(current_value: &mut Option<Self::Item>, remove_value: Self::Item) {
+                    match current_value {
+                        Some(current) if *current == remove_value => *current_value = None,
+                        _ => {}
+                    }
+                }
+            }
+        )*
+    }
+}
+
+append_remove_for_single_value!(
+    u32,
+    String,
+    Utf8PathBuf,
+    ConnectUrl,
+    bool,
+    IpAddr,
+    NonZeroU16,
+    u16
+);
```

Comment:
As an alternative to my previous suggestion of moving the trait into the `tedge_config_macros` crate, we could avoid the duplication by providing a stub implementation here as the relevant code will never be called.

```suggestion
impl<T> AppendRemoveItem for T {
    type Item = T;
    
    fn append(_current_value: &mut Option<Self::Item>, _new_value: Self::Item) {
        unimplemented!()
    }
    
    fn remove(_current_value: &mut Option<Self::Item>, _remove_value: Self::Item) {
        unimplemented!()
    }
}
```